### PR TITLE
[WEBSITE-890] Initial implementation of multi_direction_sort

### PR DESCRIPTION
**Pull request by katylouise**

### CODE REVIEW

File: lib/parliament/ntriple/utils.rb [hunk 1]
```diff
@@ -49,6 +49,7 @@ def self.sort_by(args)
         args[:prepend_rejected] ? rejected.concat(list) : list.concat(rejected)
       end
 
+
```
Comment (mattrayner) [inline lib/parliament/ntriple/utils.rb]: remove this 😛

File: lib/parliament/ntriple/utils.rb [hunk 2]
```diff
@@ -94,6 +95,27 @@ def self.sort_defaults
         { prepend_rejected: true }
       end
 
+      def self.multi_direction_sort(args)
```
Comment (mattrayner) [inline lib/parliament/ntriple/utils.rb]: Can we add some documentation to this including examples?

File: lib/parliament/ntriple/utils.rb [hunk 3]
```diff
@@ -94,6 +95,27 @@ def self.sort_defaults
...
+        directions_hash = { asc: 1, desc: -1 }
+        rejected = []
+        args = sort_defaults.merge(args)
+
+        list = args[:list].dup
+        sort_directions = args[:parameters]
+
+        list, rejected = prune_list(list, rejected, sort_directions.keys)
+        list.sort! do |obj1, obj2|
```
Comment (mattrayner) [inline lib/parliament/ntriple/utils.rb]: We need to look at the efficiency of this algorithm - do we understand the potential complexity?

File: lib/parliament/response/ntriple_response.rb [hunk 4]
```diff
@@ -107,6 +107,15 @@ def sort_by(*parameters)
         )
       end
 
+      def multi_direction_sort(parameters)
```
Comment (mattrayner) [inline lib/parliament/response/ntriple_response.rb]: Documentation including examples is also required here

File: spec/parliament/ntriple/utils_spec.rb [hunk 5]
```diff
@@ -184,4 +184,24 @@
       expect(sorted_incumbencies[1].end_date).to eq(DateTime.new(2015, 3, 30))
     end
   end
+
+  describe '#multi_direction_sort' do
+    it 'returns a response sorted by member_count (desc) and name (asc)' do
```
Comment (mattrayner) [inline spec/parliament/ntriple/utils_spec.rb:189]: We need to add tests for (asc), (asc) - (desc), (desc) + what if you provide more that two?

File: spec/parliament/response/ntriple_response_spec.rb [hunk 6]
```diff
@@ -161,4 +161,23 @@
       expect(sorted_response[1].incumbencyStartDate).to eq('2010-05-06')
     end
   end
+
+  describe '#multi_direction_sort' do
```
Comment (mattrayner) [inline spec/parliament/response/ntriple_response_spec.rb:165]: See above

**Review by mattrayner — CHANGES_REQUESTED** [review]

File: spec/parliament/ntriple/utils_spec.rb (hunk 5 above)
Comment (katylouise) [reply spec/parliament/ntriple/utils_spec.rb:189]: Done - there are now tests for asc asc, desc desc, desc asc and three parameters.

Comment (mattrayner) [reply spec/parliament/ntriple/utils_spec.rb:189]: Just to check - is there an upper limit to the number of sorts and orders you will allow?

If so we should raise an error if more than that - if not we should document and maybe have like a 10 deep test for sanity checking

**Review by mattrayner — CHANGES_REQUESTED** [review]
See comment relating to upper limit of multi_order sort

File: spec/parliament/ntriple/utils_spec.rb (hunk 5 above)
Comment (katylouise) [reply spec/parliament/ntriple/utils_spec.rb:189]: Done - there is now an extra test for 5 sort parameters

**Review by mattrayner — APPROVED** [review]

**Pull request merged**